src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/TextMapPropagator.cs

```diff
@@ -0,0 +1,520 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Text;
+using System.Net;
+using System.Collections.Generic;
+
+namespace System.Diagnostics
+{
+    public abstract class TextMapPropagator
+    {
+        public delegate bool PropagatorGetterCallback(object carrier, string fieldName, out string? value);
+
+        public abstract IReadOnlyCollection<string> Fields { get; }
+
+        // Inject
+
+        public abstract bool Inject(Activity activity, object carrier, Action<object /* carrier */, string /* field name */, string /* value to inject */> setter);
+        public abstract bool Inject(ActivityContext context, object carrier, Action<object /* carrier */, string /* field name */, string /* value to inject */> setter);
+        public abstract bool Inject(IEnumerable<KeyValuePair<string, string?>> baggage, object carrier, Action<object /* carrier */, string /* field name */, string /* value to inject */> setter);
+
+        // Extract
+
+        public abstract bool Extract(object carrier, PropagatorGetterCallback getter, out string? id, out string? state);
+        public abstract bool Extract(object carrier, PropagatorGetterCallback getter, out ActivityContext context);
+        public abstract bool Extract(object carrier, PropagatorGetterCallback getter, out IEnumerable<KeyValuePair<string, string?>>? baggage);
+
+        //
+        // Static APIs
+        //
+
+        public static TextMapPropagator DefaultPropagator { get; set; } = CreateLegacyPropagator();
+
+        // For Microsoft compatibility. e.g., it will propagate Baggage header name as "Correlation-Context" instead of "baggage".
+        public static TextMapPropagator CreateLegacyPropagator() => new LegacyTextMapPropagator();
+
+        // Suppress context propagation.
+        public static TextMapPropagator CreateOutputSuppressionPropagator() => new OutputSuppressionPropagator();
+
+        // propagate only root parent context and ignore any intermediate created context.
+        public static TextMapPropagator CreatePassThroughPropagator() => new PassThroughPropagator();
```

Comment:
Similar to the above, this PassThroughPropagator is implemented to always emit baggage using the Correlation-Context header (even if it was received via the W3C baggage header). If someone wanted a different behavior such as always emitting with the W3C baggage header or using different baggage headers in different circumstances they would need to author their own propagator. No objection from me, but I want to highlight which customizations are supported and which aren't to make sure everyone is in agreement.